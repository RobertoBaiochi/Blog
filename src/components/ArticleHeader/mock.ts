import { ArticleHeaderProps } from '.';
import { data } from '../../api/data.json';

// const { posts } = data;
// const {
//   id,
//   attributes: { title, excerpt, createdAt },
// } = posts.data[1];

// export default {
//   id,
//   title,
//   excerpt,
//   cover: data.posts.data[1].attributes.cover,
//   author: data.posts.data[1].attributes.author.data,
//   categories: data.posts.data[1].attributes.categories.data,
//   createdAt,
// } as unknown as ArticleHeaderProps;

const {
  id,
  attributes: { title, excerpt, createdAt },
} = data.posts.data[1];

export default {
  id,
  title,
  excerpt,
  cover: data.posts.data[1].attributes.cover,
  author: data.posts.data[1].attributes.author.data,
  categories: data.posts.data[1].attributes.categories.data[1],
  createdAt,
} as unknown as ArticleHeaderProps;
