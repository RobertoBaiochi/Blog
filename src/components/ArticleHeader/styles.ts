import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.header`
  ${({ theme }) => css`
    padding-bottom: ${theme.spacing.xlarge};
    margin-bottom: ${theme.spacing.xlarge};
    border-bottom: 0.1rem solid ${theme.colors.mediumGray};

    ${HeadingStyles} {
      margin: 0;
      margin-bottom: ${theme.spacing.small};
    }
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: ${theme.spacing.medium} 0;
    font-size: ${theme.font.sizes.medium};
  `};
`;

export const Cover = styled.img`
  ${({ theme }) => css`
    display: block;
    margin-bottom: ${theme.spacing.medium};
  `};
`;
