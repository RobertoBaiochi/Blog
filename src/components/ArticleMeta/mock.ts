import { ArticleMetaProps } from './';
import { data } from '../../api/data.json';

const { posts } = data;

const {
  attributes: {
    createdAt,
    author: {
      data: {
        id: idAuthor,
        attributes: { displayName: displayNameAuthor, slug: slugAuthor },
      },
    },
    categories,
  },
} = posts.data[1];

export default {
  createdAt,
  author: {
    idAuthor,
    displayNameAuthor,
    slugAuthor,
  },
  categories,
} as unknown as ArticleMetaProps;

// import { ArticleMetaProps } from '.';

// export default {
//   createdAt: '2023-05-09T04:49:38.694Z',
//   author: {
//     id: '1',
//     displayName: 'Roberto Baiochi',
//     slug: 'roberto-baiochi',
//   },
//   categories: [
//     {
//       id: '1',
//       displayName: 'ReactJs',
//       slug: 'react-js',
//     },
//     {
//       id: '2',
//       displayName: 'NextJs',
//       slug: 'next-js',
//     },
//   ],
// } as ArticleMetaProps;
