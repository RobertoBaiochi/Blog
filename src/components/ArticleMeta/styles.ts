import styled, { css } from 'styled-components';

export const Wrapper = styled.div`
  ${() => css`
    width: 100%;
  `}
`;
