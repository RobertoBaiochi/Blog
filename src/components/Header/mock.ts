import { HeaderProps } from '.';
import { data } from '../../api/data.json';

const { setting } = data;
const {
  data: {
    id,
    attributes: { blogName, blogDescription },
  },
} = setting;

export default {
  id,
  blogName,
  blogDescription,
  logo: data.setting.data.attributes.logo.data,
} as unknown as HeaderProps;
