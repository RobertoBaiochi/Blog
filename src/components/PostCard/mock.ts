import { PostCardProps } from '.';
import { data } from '../../api/data.json';

export default data.posts.data[1] as unknown as PostCardProps;
