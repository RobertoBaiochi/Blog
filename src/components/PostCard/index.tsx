import Link from 'next/link';
import { Cover, Excerpt, Wrapper } from './styles';
import Heading from '../Heading';
import { StrapiImage } from '@/shared-typed/cover';

export type PostCardProps = {
  id: number;
  attributes: {
    title: string;
    slug: string;
    excerpt: string;
    cover: StrapiImage;
  };
};

const PostCard = ({ attributes }: PostCardProps) => {
  const { cover, title, slug, excerpt } = attributes;
  const {
    attributes: { alternativeText, url },
  } = cover.data[0];

  return (
    <Wrapper>
      <Link href={`/post/${slug}`}>
        <Cover src={url} alt={alternativeText} />
      </Link>

      <Heading as="h2" size="small">
        <Link href={`/post/${slug}`}>{title}</Link>
      </Heading>

      <Excerpt>{excerpt}</Excerpt>
    </Wrapper>
  );
};
export default PostCard;
