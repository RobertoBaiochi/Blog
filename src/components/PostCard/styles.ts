import styled, { css } from 'styled-components';
import { Title as HeadingStyles } from '../Heading/styles';

export const Wrapper = styled.div`
  ${({ theme }) => css`
    text-decoration: none;

    ${HeadingStyles} {
      margin: 0;
      margin: ${theme.spacing.small} 0;
      padding-inline: ${theme.spacing.xsmall};
    }

    a {
      text-decoration: none;
      color: inherit;
      transition: all 300ms ease-in-out;
    }

    @media( hover: hover) {
      &:hover a {
        color: ${theme.colors.secondary};
      }

      &:hover img {
        opacity: 0.8;
      }
    }

  `}
`;

export const Cover = styled.img`
  ${() => css`
    width: 100%;
    height: 200px;
    transition: all 300ms ease-in-out;
  `}
`;

export const Excerpt = styled.p`
  ${({ theme }) => css`
    margin: 0;
    color: ${theme.colors.darkText};
    padding-inline: ${theme.spacing.xsmall};
  `};
`;
