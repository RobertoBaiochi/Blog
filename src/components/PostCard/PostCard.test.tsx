import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import PostCard, { PostCardProps } from '.';
import '@testing-library/jest-dom';
import mock from './mock';

const props: PostCardProps = mock;

describe('<PostCard />', () => {
  it('should render heading, cover and excerpt', () => {
    renderTheme(<PostCard {...props} />);
    const {
      attributes: { cover, excerpt, slug, title },
    } = props;
    const {
      attributes: { alternativeText },
    } = cover.data[0];

    expect(screen.getByRole('heading', { name: title })).toBeInTheDocument();
    expect(screen.getByAltText(alternativeText)).toBeInTheDocument();
    expect(screen.getByText(excerpt)).toBeInTheDocument();
    expect(screen.getByRole('link', { name: title })).toHaveAttribute(
      'href',
      `/post/${slug}`,
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<PostCard {...props} />);

    expect(container).toMatchSnapshot();
  });
});
