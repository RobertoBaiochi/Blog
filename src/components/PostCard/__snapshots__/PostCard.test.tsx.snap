// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCard /> should match snapshot 1`] = `
<div>
  <div
    class="sc-12652a66-0 epxRKc"
  >
    <a
      href="/post/usando-o-canvas-para-criar-graficos-em-java-script"
    >
      <img
        alt="praia vista de cima, água cor vende escuro e rochas"
        class="sc-12652a66-1 iYcwyv"
        src="https://res.cloudinary.com/dfmbf9sae/image/upload/v1683581086/adam_seckel_xhaz6_Nn_VT_64_unsplash_ad1c1db2e6.jpg"
      />
    </a>
    <h2
      class="sc-b5aaf39e-0 kQHlLj"
    >
      <a
        href="/post/usando-o-canvas-para-criar-graficos-em-java-script"
      >
        Usando o Canvas para criar gráficos em JavaScript
      </a>
    </h2>
    <p
      class="sc-12652a66-2 jhiHKy"
    >
      O artigo apresenta como usar a API Canvas em HTML e JavaScript para criar gráficos personalizados com estilização e interatividade, resultando em visualizações de dados impressionantes e envolventes.
    </p>
  </div>
</div>
`;
